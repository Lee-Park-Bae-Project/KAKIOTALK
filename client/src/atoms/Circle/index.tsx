import React, { FC } from 'react';
import styled from 'styled-components';
import { color, weight, fontSize } from 'styles/global';

interface Props{
  num: number;
}

const S = {
  Container: styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: fit-content;
    height: 1rem;
    margin: 1rem;
    padding: 0.1rem 0.5rem;
    border-radius: 2rem;
    text-align: center;
    font-size: ${fontSize.SMALL};
    background-color: ${color.RED};
    color: ${color.WHITE};
    font-weight: ${weight.NORMAL};
  `,
};
const Circle: FC<Props> = ({
  num,
}) => {
  const newNum = Math.log10(num) > 2
    ? '999+'
    : num.toString();
  return (
    <S.Container>
      {newNum}
    </S.Container>
  );
};

export default Circle;
