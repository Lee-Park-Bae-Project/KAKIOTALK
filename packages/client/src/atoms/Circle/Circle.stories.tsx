import React from 'react'
import Circle from 'atoms/Circle'

export default {
  title: 'Atoms/Circle',
  component: Circle,
}

export const Default = () => {
  const num = 1

  return (
    <Circle num={num}/>
  )
}

export const BiggerNum = () => {
  const num = 33

  return (
    <Circle num={num}/>
  )
}

export const BiggerBiggerNum = () => {
  const num = 1000

  return (
    <Circle num={num}/>
  )
}
