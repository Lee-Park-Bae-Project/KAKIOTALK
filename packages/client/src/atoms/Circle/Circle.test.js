/* eslint-disable no-undef */
import React from 'react'
import { mount } from 'enzyme'
import Circle from 'atoms/Circle'

describe('<Circle />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Circle num={22}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches props', () => {
    const wrapper = mount(<Circle num={22}/>)
    expect(wrapper.props().num).toBe(22)
  })
  it('matches big number converted with ...', () => {
    const wrapper = mount(<Circle num={9999}/>)
    expect(wrapper.find('div').text()).toBe('999+')
  })
})
