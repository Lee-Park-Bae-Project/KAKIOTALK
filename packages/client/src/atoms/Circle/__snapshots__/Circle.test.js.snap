// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Circle /> matches snapshot 1`] = `
<Circle
  num={22}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "fJTehO",
            "rules": Array [
              "
    display: flex;
    align-items: center;
    justify-content: center;
    width: fit-content;
    height: 1rem;
    margin: 1rem;
    padding: 0.1rem 0.5rem;
    border-radius: 2rem;
    text-align: center;
    font-size: ",
              "0.8rem",
              ";
    background-color: ",
              "#DD6F5A",
              ";
    color: ",
              "#FEFFFF",
              ";
    font-weight: ",
              "400",
              ";
  ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa fJTehO"
      >
        22
      </div>
    </StyledComponent>
  </styled.div>
</Circle>
`;
